slices.go

```diff
@@ -99,3 +99,26 @@ func Filter[T any](s []T, f func(T) bool) []T {
 
 	return n
 }
+
+// Sort function
+func Sort[T any](s []T, fn func(T, T) bool) []T {
+	temp := make([]T, len(s))
+	copy(temp, s) // Copy slice because this is a reference
+
+	var n []T // Init return slice
```

Comment:
How about this? @rmacciuci 
var n = make([]T, 0, len(s))

Reply:
Hi, is a goog idea, I will change the code and I will update this PR